Modify tasks to use systemd or freshclam binary

ClementJ35 opened the pull request:
## 🗣 Description ##

As describe in  #38, setup based on OSes doesn't make sense now as many system use systemd.
It's better to choose using the fact : ansible_service_mgr

## 🧪 Testing ##

Tested on Debian 11 & CentOS 8

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

ClementJ35 committed:
[tasks] Setup bissect systemd / not systemd

ClementJ35 commented:
Failing test seems imputable to docker instance : 
'''2022-01-31T17:01:49.4122921Z fatal: [fedora32]: FAILED! => {"changed": false, "msg": "failure 1 during daemon-reload: System has not been booted with systemd as init system (PID 1). Can't operate.\nFailed to connect to bus: Host is down\n"}'''
In the real world, Fedora 32 is systemd based.
Either ansible facts are wrong about systemd or the docker image have some ghosts information.

jsf9k commented on tasks/main.yml:
```diff
@@ -35,20 +35,13 @@
+  when: ansible_service_mgr == 'systemd'
+
+- name: Load tasks file for bin setup
+  ansible.builtin.include_tasks: setup_bin.yml
```
I think we should consider renaming `setup_bin.yml` to `setup_manual.yml`.  The latter more clearly indicates the purpose of the file (IMO).
```suggestion
- name: Load tasks file for manual setup
  ansible.builtin.include_tasks: setup_manual.yml
```

dav3r replied:
We also need to add some sort of documentation about this file and what it should contain.

ClementJ35 replied:
I did the changes mentioned above in [e485997](https://github.com/cisagov/ansible-role-clamav/pull/39/commits/e4859970e9b3a4cbaa125b75501620b9e038f4ac).
Can you check it's all right ?

jsf9k replied:
Looks good to me.  Thanks for making the changes.  @dav3r and @mcdonnnj, can you guys take another look?

dav3r replied:
I'm satisfied with the recent changes for this - thanks.

mcdonnnj replied:
The updates work for me as well.

jsf9k reviewed (changes requested):
Thanks for another great improvement, @ClementJ35!  Please consider my change request while I fix the failing builds.

jsf9k committed:
Add Fedora 35, remove Fedora 32 and 33

Fedora 35 has been released, and Fedora 32 and 33 are EOL.

This change would eventually be made via
cisagov/skeleton-ansible-role#95 anyway.  Making it now fixes a build
issue:
https://github.com/cisagov/ansible-role-clamav/runs/5009381037?check_suite_focus=true.

jsf9k committed:
Adjust test to also include Fedora 35

The Fedora 34 check can be dropped, since Fedora 32 and 33 are no
longer supported.

jsf9k commented:
> Failing test seems imputable to docker instance : '''2022-01-31T17:01:49.4122921Z fatal: [fedora32]: FAILED! => {"changed": false, "msg": "failure 1 during daemon-reload: System has not been booted with systemd as init system (PID 1). Can't operate.\nFailed to connect to bus: Host is down\n"}''' In the real world, Fedora 32 is systemd based. Either ansible facts are wrong about systemd or the docker image have some ghosts information.

I think Fedora 32 lacks the ClamAV SystemD service that is being manipulated in this role, even though it does run SystemD.

ClementJ35 committed:
Change name of manual task and add comments

jsf9k reviewed (approved):
Thanks for simplifying this Ansible role!

dav3r reviewed (approved):
This looks great to me now! 👍 👍

mcdonnnj reviewed (approved):
Strong work! 💪💪💪
